shell/src/main/java/alluxio/cli/LogLevel.java

```diff
@@ -0,0 +1,193 @@
+    System.err.println(message);
+    HelpFormatter help = new HelpFormatter();
+    help.printHelp(USAGE, OPTIONS);
+  }
+
+  /**
+   * Implements log level setting and getting.
+   *
+   * @param args list of arguments
+   * @return 0 on success, 1 on failures
+   */
+  public static int logLevel(String... args) {
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd;
+
+    try {
+      cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+    } catch (ParseException e) {
+      printHelp("Unable to parse input args: " + e.getMessage());
+      return 1;
+    }
+    Preconditions.checkNotNull(cmd, "Unable to parse input args");
+    List<TargetInfo> addrList = new ArrayList<>();
+    String logName = "";
+    String level = null;
+    String[] targets;
+
+    String argName = cmd.getOptionValue(LOGNAME_OPTION_NAME);
+    if (StringUtils.isNotBlank(argName)) {
+      logName = argName;
+    }
+
+    if (cmd.hasOption(TARGET_OPTION_NAME)) {
+      String argTarget = cmd.getOptionValue(TARGET_OPTION_NAME);
+      if (StringUtils.isBlank(argTarget)) {
+        targets = new String[]{"master", "workers"};
+      } else if (argTarget.contains(",")) {
+        targets = argTarget.split(",");
+      } else {
+        targets = new String[]{argTarget};
+      }
+    } else {
+      targets = new String[]{"master", "workers"};
+    }
+    for (String target : targets) {
+      if (target.contains("master")) {
+        String masterHost = NetworkAddressUtils.getConnectHost(ServiceType.MASTER_WEB);
+        int masterPort = NetworkAddressUtils.getPort(ServiceType.MASTER_WEB);
+        addrList.add(new TargetInfo(masterHost + ":" + masterPort, "master"));
+      } else if (target.contains("workers")) {
+        AlluxioBlockStore alluxioBlockStore = AlluxioBlockStore.create();
+        try {
+          List<BlockWorkerInfo> workerInfoList = alluxioBlockStore.getWorkerInfoList();
+          for (BlockWorkerInfo workerInfo : workerInfoList) {
+            WorkerNetAddress netAddress = workerInfo.getNetAddress();
+            addrList.add(
+                new TargetInfo(netAddress.getHost() + ":" + netAddress.getWebPort(), "worker"));
+          }
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      } else if (target.contains(":")) {
+        addrList.add(new TargetInfo(target, "worker"));
+      }
+    }
+
+    if (cmd.hasOption(LEVEL_OPTION_NAME)) {
+      String argLevel = cmd.getOptionValue(LEVEL_OPTION_NAME);
+      if (StringUtils.isNotBlank(argLevel)) {
+        level = argLevel;
+      }
+    }
+
+    for (TargetInfo targetInfo : addrList) {
+      String url = "http://" + targetInfo.getAddres() + Constants.REST_API_PREFIX + "/"
+          + targetInfo.getRole() + "/" + LOG_LEVEL + "?" + LOGNAME_OPTION_NAME + "=" + logName;
+      if (level != null) {
+        url += "&" + LEVEL_OPTION_NAME + "=" + level;
+      }
+      String result = HttpUtils.sendPost(url, 5000);
```

Comment:
instead of reading the whole body first and then parse it as json, a preferred way is to parse the response body as a json stream.